AttributeError: 'Subscript' object has no attribute 'name'

Seems distinct from https://github.com/PyCQA/pylint/issues/2250
Maybe a regression, or another edge case?

### Steps to reproduce

Given a file `a.py`:
```python
from attr import attrs, attrib
from typing import Optional

@attrs()
class User:
    name: str = attrib()
    age: int = attrib()
    occupation = Optional[str] = attrib(default=None)
```
Where a typo was made in the last line - first `=` should be `:`.

### Current behavior

Pylint crashes if executed on the file, or on any file that imports it.

```
# pylint a.py
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1041, in get_ast
    return MANAGER.ast_from_file(filepath, modname, source=True)
  File "/usr/local/lib/python3.7/site-packages/astroid/manager.py", line 98, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
  File "/usr/local/lib/python3.7/site-packages/astroid/builder.py", line 142, in file_build
    return self._post_build(module, encoding)
  File "/usr/local/lib/python3.7/site-packages/astroid/builder.py", line 166, in _post_build
    module = self._manager.visit_transforms(module)
  File "/usr/local/lib/python3.7/site-packages/astroid/manager.py", line 75, in visit_transforms
    return self._transform.visit(node)
  File "/usr/local/lib/python3.7/site-packages/astroid/transforms.py", line 90, in visit
    module.body = [self._visit(child) for child in module.body]
  File "/usr/local/lib/python3.7/site-packages/astroid/transforms.py", line 90, in <listcomp>
    module.body = [self._visit(child) for child in module.body]
  File "/usr/local/lib/python3.7/site-packages/astroid/transforms.py", line 58, in _visit
    return self._transform(node)
  File "/usr/local/lib/python3.7/site-packages/astroid/transforms.py", line 41, in _transform
    ret = transform_func(node)
  File "/usr/local/lib/python3.7/site-packages/astroid/brain/brain_attrs.py", line 58, in attr_attributes_transform
    node.locals[target.name] = [rhs_node]
AttributeError: 'Subscript' object has no attribute 'name'
************* Module a
a.py:1:0: F0002: <class 'AttributeError'>: 'Subscript' object has no attribute 'name' (astroid-error)
```

**Additionally:** If you run `pylint b.py` where `b.py` is:
```python
from a import User
```
The error message does not mention `a.py` at all:
```
Exception on node <ImportFrom l.3 at 0x7fe42b22d490> in file '/tmp/b.py'
Traceback (most recent call last):
<SNIP>
AttributeError: 'Subscript' object has no attribute 'name'
```

### Expected behavior

Should catch and report the problem, not crash. Perhaps:
```
************* Module a
a.py:8:13: E1137: 'Optional' does not support item assignment (unsupported-assignment-operation)
```

### pylint --version output

Result of `pylint --version` output:

```
# pylint --version
pylint 3.0.0a3
astroid 2.5.6
Python 3.7.7 (default, Apr 16 2020, 18:55:06)
[GCC 8.3.0]
```

Additional dependencies
```
attrs==19.3.0
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -17,6 +17,11 @@ Release date: TBA
 ..
   Put bug fixes that should not wait for a new minor version here
 
+* Fix a crash when there would be a 'TypeError object does not support
+  item assignment' in the code we parse.
+
+  Closes #4439
+
 * Fix crash if a callable returning a context manager was assigned to a list or dict item
 
   Closes #4732

--- a/requirements_test_min.txt
+++ b/requirements_test_min.txt
@@ -1,4 +1,4 @@
 -e .
-astroid==2.6.4  # Pinned to a specific version for tests
+astroid==2.6.5  # Pinned to a specific version for tests
 pytest~=6.2
 pytest-benchmark~=3.4

--- a/setup.cfg
+++ b/setup.cfg
@@ -42,7 +42,7 @@ project_urls =
 [options]
 packages = find:
 install_requires =
-    astroid>=2.6.4,<2.7 # (You should also upgrade requirements_test_min.txt)
+    astroid>=2.6.5,<2.7 # (You should also upgrade requirements_test_min.txt)
     isort>=4.2.5,<6
     mccabe>=0.6,<0.7
     toml>=0.7.1

--- a/tests/functional/r/regression/regression_4439.py
+++ b/tests/functional/r/regression/regression_4439.py
@@ -1,0 +1,13 @@
+"""AttributeError: 'Subscript' object has no attribute 'name' """
+# pylint: disable=missing-docstring
+
+from typing import Optional
+
+from attr import attrib, attrs
+
+
+@attrs()
+class User:
+    name: str = attrib()
+    age: int = attrib()
+    occupation = Optional[str] = attrib(default=None) # [unsupported-assignment-operation]

--- a/tests/functional/r/regression/regression_4439.txt
+++ b/tests/functional/r/regression/regression_4439.txt
@@ -1,0 +1,1 @@
+unsupported-assignment-operation:13:17:User:'Optional' does not support item assignment:HIGH

